config/config.go

```
@@ -36,6 +37,37 @@ type Config struct {
 	Blockchain       *BlockchainConfig
 }
 
+func (c *Config) ProvideNodeKey(key string, password string) error {
+	instanceDir := filepath.Join(c.DataDir, "keystore")
+	if err := os.MkdirAll(instanceDir, 0700); err != nil {
+		return err
+	}
+
+	keyfile := filepath.Join(instanceDir, datadirPrivateKey)
+	if _, err := crypto.LoadECDSA(keyfile); err == nil {
+		return errors.New("key already exists")
+	}
+
+	keyBytes, err := hex.DecodeString(key)
+	if err != nil {
+		return errors.New("error while decoding key")
```

Don't hide an original error